gdk/common/validate_user_input_recipe.py

```diff
@@ -0,0 +1,84 @@
+import json
+import logging
+import sys
+from pathlib import Path
+
+import gdk.common.consts as consts
+import gdk.common.exceptions.error_messages as error_messages
+import gdk.common.utils as utils
+
+
+def get_recipe():
+    """
+    Loads the component's recipe from the user-input recipe file as a python object.
+
+    Parameters
+    ----------
+        None
+
+    Returns
+    -------
+        recipe_data: component's recipe as a dictionary object if the user-input recipe file is valid.
+    """
+    user_recipe_file = get_user_input_recipe_file()
+
+    # validate the json recipe syntax
+    recipe_data = validate_recipe_syntax(user_recipe_file)
+
+    return recipe_data
+
+
+def get_user_input_recipe_file():
```

Comment:
This piece here is redundant as we've an existing method to get the recipe file path in the GDKProject - https://github.com/aws-greengrass/aws-greengrass-gdk-cli/blob/acaa624fae82bc8faf9e9ab726b710bdad0cd8bf/gdk/common/config/GDKProject.py#L11

Reply:
Nit: IMO, this function should not be responsible for finding the recipe file path as that may not be the same every time. Eg. Build recipe file exists at the path `greengrass-build/recipes/recipe.yaml`. So, if we want to validate the build recipe file in future, this function won't be helpful.

We can rewrite the recipe validation in such a way that it can be extensible to the recipe path at any path. Something like

```
recipe_validator = RecipeValidator("/path/to/recipe")
recipe_validator.validate_syntax()
recipe_validator.validate_semantics() 
```